docs/content/guides/developer/getting-started/data-serving.mdx

```diff
@@ -0,0 +1,135 @@
+| September 2025 | Deprecation of JSON-RPC | **JSON-RPC will be deprecated at this point and migration notice period will start.** |
+| December 2025 | GA release of GraphQL RPC Server and Indexer 2.0 | You can start migration and cutover of your application in the production environment. |
+| June 2026 | End of migration timeline | **JSON-RPC will be fully deactivated at this point.** This timeline assumes about 9 months of migration notice period. |
+
+See [below](#when-to-use-grpc-vs-graphql-with-indexer-20) for a comparison with the gRPC API.
+
+### When to use gRPC vs GraphQL with Indexer 2.0
+
+You can use the high-level criteria mentioned in the following table to determine whether gRPC API or GraphQL RPC with Indexer 2.0 would better serve your use case. It's not an exhaustive list and it's expected that either of the options could work suitably for some of the use cases.
+
+| **Dimension** | **gRPC API** | **GraphQL RPC with Indexer 2.0** |
+| -------- | ------- | ------- |
+| Type of application or data consumer | Ideal for Web3 exchanges, defi market maker apps, other defi protocols or apps with ultra low-latency needs | Ideal for webapp builders or builders with slightly relaxed latency needs |
+| Query patterns | Okay to read data from different endpoints separately and combine on the client-side; faster serialization, parsing and validation due to binary format | Need easier decoupling of the client with the ability to combine different types of data in fewer calls; data will be consistent across the underlying queries, including for paginated results |
+| Retention period requirements | Default retention period will be 2 weeks with actual configuration dependent on the Full node operator and their needs & goals; see history-related note after the table | Default retention period in postgres database will be 4 weeks with actual configuration depending on your or a data indexer operator's needs; see history-related note after the table |
+| Streaming needs | Will include a streaming or subscription API before beta release | Subscription API is planned but will be available after GA |
+| Incremental costs | Little to no incremental costs if already using Full node JSON-RPC | Somewhat significant incremental costs if already using Full node JSON-RPC and if retention period & query patterns differences are insignificant |
+
+:::note
+
+Above table only mentions the default retention period for both options. The expectation is that it should be reasonable for you or a Full node / data indexer operator to configure that to a few times higher without impacting the performance significantly, but relevant guidelines will be shared before each option's respective GA release. Those will include recommendations for how to access historical data beyond the configured retention period.
```

Comment:
This note doesn't really capture the fact that GraphQL *can* talk to KV store -- i.e. it is an interface for the archival store, while gRPC will not be.

Reply:
I've tweaked the language to reflect that. Though as an internal note, we do need to figure out a good mechanism for historical data access for folks using gRPC :)